### PR TITLE
feat: remove gossip code

### PR DESCRIPTION
**Pull request by joshuef**
BREAKING CHANGE: remove gossip as royalties will be collected via DAG

## Description

reviewpad:summary

### CODE REVIEW

File: sn_node/src/event.rs [hunk 1]
```diff
@@ -67,14 +67,6 @@ pub enum NodeEvent {
...
-    /// Gossipsub message received
-    GossipsubMsg {
-        /// Topic the message was published on
-        topic: String,
-        /// The raw bytes of the received message
-        #[debug(skip)]
-        msg: Bytes,
-    },
     /// Transfer notification message received for a public key
     TransferNotif {
```
Comment (grumbach) [inline sn_node/src/event.rs]: I think `TransferNotif` is also related to gossip

File: sn_node_rpc_client/src/main.rs [hunk 2]
```diff
@@ -67,26 +64,6 @@ enum Cmd {
         #[command(flatten)]
         peers: PeersArgs,
     },
-    /// Subscribe to a given Gossipsub topic
```
Comment (grumbach) [inline sn_node_rpc_client/src/main.rs:70]: `TransfersEvents {` just above this is also part of gossip

**Review by grumbach — APPROVED** [review]
LGTM! Spotted a couple of remaining zombie types, but looking good!

**Comment by joshuef** [comment]
Sorted that. Merged it in,

**Pull request merged**